bot/cogs/spam/spam.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pathlib
+import time
+
+import discord
+from discord.ext import commands
+
+from ..variables import *
+
+spam_log_file = str(
+    pathlib.Path(
+        os.path.join(
+            os.path.abspath(os.getcwd()), "bot", "cogs", "spam", "spam_log.txt"
+        )
+    ).resolve()
+)
+
+
+class Spam(commands.Cog):
+    def __init__(self, client):
+        self.client = client
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        if message.author.id == self.client.user.id:
+            return
+
+        ctx = await self.client.get_context(message)
+
+        _timestamp = str(int(time.time()))
+        _authorid = str(message.author.id)
+        _channelid = str(message.channel.id)
+
+        _log = _timestamp + "___" + _authorid + "___" + _channelid
+
+        counter_messages = 0
+        counter_channels = 0
+        list_channels = []
+
+        with open(spam_log_file, "r+") as file:
+            file.writelines(f"{str(_log)}\n")
+
+            for line in file.readlines():
+                log_split = line.split("___")
+                log_timestamp = log_split[0]
+                log_authorid = log_split[1]
+                log_channelid = log_split[2]
+
+                # total number of messages sent by the user, in the last X seconds
+                if log_authorid == str(message.author.id):
+                    counter_messages += 1
+                    list_channels.append(log_channelid)  # add channel id
```

Comment:
The X seconds condition is only in the comment. Meaning that if I say "hello" three times in the lifetime of the log (e.g. a month) I will get banned.

Furthermore we flag here on all messages and never compare if the message are actually duplicates.

Reply:
Update: I now see the clearing of the log outside of this cog..

I guess this logic actually works if the interval is low enough (e.g. 1 second)

Reply:
Yes, I tested with 2-5 seconds and it seemed reasonable